### DISCUSSION TITLE
Crosshair mod

### DISCUSSION THREAD

**Discussion by silver-volt4** [ppy/osu#13053] [Ideas] [answered]
**Describe the new feature:** 
An unranked, entirely for fun mod that could be added to the game. Rather than moving the cursor to click circles, the cursor will be always in the center of the screen, and mouse movement will move the position of the circles (so they are under the actual cursor in the center, and then scored by clicking). This could be used by people to practice for shooter games.

**Proposal designs of the feature:** 
Add a new unranked mod to the mods screen, possibly to the pink section at the bottom. It could have a crosshair icon.

**Comment by silver-volt4** [comment]
I've made a short video of how it could look. (It's all a modified video. Not modified game code.)
[YouTube video](https://youtu.be/q_TmdXn8OXs)

**Comment by repsanims** [comment]
**It will have some problems:**
1. If your crosshair is in on the top left and a circle appeared in the bottom right, you will be likely to miss it.
2. People will get dizzy with the background enabled.
3. Shooter games have a 3D view and osu! have a 2D view.

**The solution:**
You can create a much bigger yet less opaque approaching circle, and/or an arrow indicator on where the circle appeared
It can be a fixed background, like aim training map in shooter games.
And maybe someone can create a custom shader that makes the view distorted and had an inverted fisheye (like) view.

**Another feature:**
To match the quality of shooter games, the score is multiplied based on how center your crosshair is.

Shooter games like CSGO have more target horizontally with a small amount of vertical difference, so maybe spread the circle more horizontally and less vertically
And if you see pro-players play shooter games, they can shoot an enemy even on the edge of the screen, so maybe increase the circle's distance.

_**Unrelated article**_
There's a fully-featured osu!mania beatmap player map in CSGO [here](https://steamcommunity.com/sharedfiles/filedetails/?id=1539561817)

**Comment by frenzibyte** [comment]
> 1. If your crosshair is in on the top left and a circle appeared in the bottom right, you will be likely to miss it.
> 2. People will get dizzy with the background enabled.
> 3. Shooter games have a 3D view and osu! have a 2D view.

1. Relative input should be recommended for this.
2. Disable it?
3. I don't see your point here

**Comment by mk56-spn** [comment, marked as answer]
https://github.com/ppy/osu/pull/19791 slightly different since it has delay ( optional ) but effectively closes this

**Reply by silver-volt4** [reply]
i completely forgot this exists lmfao
